Tasks/Common/AzureRmDeploy-common/azure-arm-rest/azure-arm-endpoint.ts

```diff
@@ -28,53 +28,63 @@ export class AzureRMEndpoint {
-                    tl.debug('certificate spn endpoint');
-                    this.endpoint.servicePrincipalCertificate = tl.getEndpointAuthorizationParameter(this._connectedServiceName, 'servicePrincipalCertificate', false);
-                    this.endpoint.servicePrincipalCertificatePath = certFilePath;
-                    fs.writeFileSync(this.endpoint.servicePrincipalCertificatePath, this.endpoint.servicePrincipalCertificate);
+            let endpointAuthScheme = tl.getEndpointAuthorizationScheme(this._connectedServiceName, true);
+            if (endpointAuthScheme && endpointAuthScheme.toLowerCase() == constants.AzureRmEndpointAuthenticationScheme.PublishProfile) {
+                this.endpoint = {
+                    subscriptionName: tl.getEndpointDataParameter(this._connectedServiceName, 'subscriptionname', true),
+                    tenantID: tl.getEndpointAuthorizationParameter(this._connectedServiceName, 'tenantid', false),
+                    scheme: endpointAuthScheme,
+                    PublishProfile: tl.getEndpointAuthorizationParameter(this._connectedServiceName, "publishProfile", true)
+                } as AzureEndpoint;
+            } else {
+                this.endpoint = {
+                    subscriptionID: tl.getEndpointDataParameter(this._connectedServiceName, 'subscriptionid', true),
+                    subscriptionName: tl.getEndpointDataParameter(this._connectedServiceName, 'subscriptionname', true),
+                    servicePrincipalClientID: tl.getEndpointAuthorizationParameter(this._connectedServiceName, 'serviceprincipalid', true),
+                    environmentAuthorityUrl: tl.getEndpointDataParameter(this._connectedServiceName, 'environmentAuthorityUrl', true),
+                    tenantID: tl.getEndpointAuthorizationParameter(this._connectedServiceName, 'tenantid', false),
+                    url: tl.getEndpointUrl(this._connectedServiceName, true),
+                    environment: tl.getEndpointDataParameter(this._connectedServiceName, 'environment', true),
+                    scheme: tl.getEndpointAuthorizationScheme(this._connectedServiceName, true),
+                    msiClientId:  tl.getEndpointDataParameter(this._connectedServiceName, 'msiclientId', true),
+                    activeDirectoryResourceID: tl.getEndpointDataParameter(this._connectedServiceName, 'activeDirectoryServiceEndpointResourceId', true),
+                    azureKeyVaultServiceEndpointResourceId: tl.getEndpointDataParameter(this._connectedServiceName, 'AzureKeyVaultServiceEndpointResourceId', true),
+                    azureKeyVaultDnsSuffix: tl.getEndpointDataParameter(this._connectedServiceName, 'AzureKeyVaultDnsSuffix', true),
+                } as AzureEndpoint;
+
+                this.endpoint.authenticationType =  tl.getEndpointAuthorizationParameter(this._connectedServiceName, 'authenticationType', true);
```

Comment:
is there any real change from here till line no 86 or is the diff not being shown correctly in PR?